thor/observation_source.py

```diff
@@ -0,0 +1,132 @@
+import abc
+from typing import TypeAlias
+
+import numpy as np
+import quivr as qv
+from adam_core.observations import detections, exposures
+
+from . import orbit
+
+
+class Observations:
+    """Observations represents a collection of exposures and the
+    detections they contain.
+
+    The detections may be a filtered subset of the detections in the
+    exposures.
+    """
+
+    detections: detections.PointSourceDetections
+    exposures: exposures.Exposures
+    linkage: qv.Linkage[detections.PointSourceDetections, exposures.Exposures]
+
+    def __init__(
+        self,
+        detections: detections.PointSourceDetections,
+        exposures: exposures.Exposures,
+    ):
+        self.detections = detections
+        self.exposures = exposures
+        self.linkage = qv.Linkage(
+            detections,
+            exposures,
+            left_keys=detections.exposure_id,
+            right_keys=exposures.id,
+        )
+
```

Comment:
We have already discussed this offline, but for the record, a lightweight class that wraps a quivr linkage is an excellent idea.